hosts/testagent-dev/configuration.nix

```diff
@@ -4,12 +4,85 @@
   self,
   inputs,
   pkgs,
+  config,
   ...
 }:
 let
   # Vendored in, as brainstem isn't suitable for nixpkgs packaging upstream:
   # https://github.com/NixOS/nixpkgs/pull/313643
   brainstem = pkgs.callPackage ../../pkgs/brainstem { };
+
+  mkAgent =
+    device:
+    let
+      # Temporary url for development
+      controllerUrl = "https://ghaf-jenkins-controller-joonasrautiola.northeurope.cloudapp.azure.com";
+      workDir = "/var/lib/jenkins/agents/${device}";
+
+      jenkins-connection-script = pkgs.writeScript "jenkins-connect.sh" ''
+        #!/usr/bin/env bash
+        set -eu
+        mkdir -p ${workDir}
+
+        # get agent.jar
+        if [ ! -f agent.jar ]; then
+          wget ${controllerUrl}/jnlpJars/agent.jar
```

Comment:
Put this in quotes even if URL is never expected to contain spaces.

You may want to check this entire shell script fragment with shellcheck.

Reply:
This and other similar cases fixed